src/inc/til/color.h

```diff
@@ -140,14 +140,11 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         // See https://en.wikipedia.org/wiki/Alpha_compositing#Description
         constexpr color layer_over(const color& destination) const
         {
-            const auto sourceAlpha = a / 255.0f;
-            const auto destinationAlpha = destination.a / 255.0f;
-            const auto aInverse = 1.0f - sourceAlpha;
-
+            const auto aInverse = (255 - a) / 255.0f;
             const auto resultA = a + destination.a * aInverse;
-            const auto resultR = (r * sourceAlpha + destination.r * destinationAlpha * aInverse) / resultA;
-            const auto resultG = (g * sourceAlpha + destination.g * destinationAlpha * aInverse) / resultA;
-            const auto resultB = (b * sourceAlpha + destination.b * destinationAlpha * aInverse) / resultA;
+            const auto resultR = (r * a + destination.r * destination.a * aInverse) / resultA;
+            const auto resultG = (g * a + destination.g * destination.a * aInverse) / resultA;
+            const auto resultB = (b * a + destination.b * destination.a * aInverse) / resultA;

```

Comment:
We don't need to multiply by 255 if we simply make sure that the part in front of the division is in the range `[0, 255^2]`. After dividing it by `resultA` (`[0, 255]`) we get a result in the range `[0, 255]`.